include/myst/thread.h

```diff
@@ -57,6 +57,13 @@ typedef struct
     uint64_t mask;
 } posix_sigaction_t;
 
+typedef struct myst_robust_list_head
+{
+    volatile void* volatile head;
+    long off;
+    volatile void* volatile pending;
+} myst_robust_list_head_t;
```

Comment:
I am not sure we need this structure. https://man7.org/linux/man-pages/man2/set_robust_list.2.html states:
> These system calls deal with per-thread robust futex lists.
       These lists are managed in user space: the kernel knows only
       about the location of the head of the list.

It's up to the user space, in our case, MUSL, to interpret the robust list. The kernel merely records where is the head of the list, and the size of each item in the list. We can use `void*` to represent the list, and `size_t` to represent the size.